src/Blazored.Modal/FocusTrap.razor

```diff
@@ -13,12 +13,16 @@
     private ElementReference _endFirst;
     private ElementReference _endSecond;
     private bool _shiftPressed;
+    private bool _shouldRender = false;
 
     [Parameter] public RenderFragment ChildContent { get; set; } = default!;
     [Parameter] public bool IsActive { get; set; }
 
     protected override bool ShouldRender()
-        => false;
+        => _shouldRender;
+
+    public void SetShouldRender(bool shouldRender = true)
+        => _shouldRender = shouldRender;
```

Comment:
I'd suggest a slightly different approach here:

```csharp
protected override bool ShouldRender() 
    => _shouldRender;

public void TriggerRender()
{
    _shouldRender = true;
    StateHasChanged();
    _shouldRender = false;
}
```

With this approach we're triggering a single render on the `FocusTrap` which allows the animation to happen but doesn't leave the `FocusTrap` with `_shouldRender` still `true`.

Reply:
Yup my bad, I just tried one solution and didn't think about the status later.